apis/projectcontour/v1/httpproxy.go

```diff
@@ -199,6 +199,30 @@ type Service struct {
 	UpstreamValidation *UpstreamValidation `json:"validation,omitempty"`
 	// If Mirror is true the Service will receive a read only mirror of the traffic for this route.
 	Mirror bool `json:"mirror,omitempty"`
+	// RequestHeaders defines how to add or remove headers from requests routed to this service.
+	// +optional
+	RequestHeaders *HeaderAlterations `json:"requestHeaders,omitempty"`
+	// ResponseHeaders defines how to add or remove headers from responses returned from this service.
+	// +optional
+	ResponseHeaders *HeaderAlterations `json:"responseHeaders,omitempty"`
+}
+
+// HeaderAlterations defines alterations to the headers being sent to or returned from a service.
+type HeaderAlterations struct {
+	// Add holds the header key/value pairs to add to those sent to or returned from a service.
+	// +optional
+	Add []HeaderAddition `json:"add,omitempty"`
```

Comment:
Is it possible to avoid the additional struct?
```Go
Add map[string]string `json:"add,omitempty"`
```

Reply:
Yeah, to keep with the conventions, we should keep it an additional struct, agreed.